(PE-1039) CES: Onboard SocGen `RWA008` and HVB `RWA009`

hbarcelos opened the pull request:
# Description

# Contribution Checklist

- [x] PR title starts with `(PE-<TICKET_NUMBER>)`
- [ ] Code approved
- [x] Tests approved
- [ ] CI Tests pass

# Checklist

- [x] Every contract variable/method declared as public/external private/internal
- [x] Consider if this PR needs the `officeHours` modifier override
- [x] Verify expiration (`30 days` unless otherwise specified)
- [ ] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [x] Validate all addresses used are in Goerli changelog or known
- [x] Notify any external teams affected by the spell so they have the opportunity to review
- [ ] Deploy spell to Goerli `ETH_GAS="XXX" ETH_GAS_PRICE="YYY" make deploy`
- [ ] Ensure contract is verified on `Goerli` etherscan
- [ ] Change test to use Goerli spell address and deploy timestamp
- [ ] Cast spell on Goerli `make spell="0x-deployed-spell-address" cast-spell`
- [ ] Run `make archive-spell` or `make date="YYYY-MM-DD" archive-spell` to make an archive directory and copy `Goerli-DssSpell.sol`, `Goerli-DssSpell.t.sol`, `Goerli-DssSpell.t.base.sol`, and `Goerli-DssSpellCollateralOnboarding.sol`
- [ ] `squash and merge` this PR

naszam commented on src/Goerli-DssSpell.sol:
```diff
@@ -35,18 +37,15 @@ contract DssSpellAction is DssAction, DssSpellCollateralAction {
+    //    https://ipfs.io/ipfs/QmPgPVrVxDCGyNR5rGp9JC5AUxppLzUAqvncRJDcxQnX1u
     //
 
-    function officeHours() public override returns (bool) {
```
Note: Office Hours should be Off by default on `spells-goerli`

naszam commented on src/Goerli-DssSpell.sol:
```diff
@@ -35,18 +37,15 @@ contract DssSpellAction is DssAction, DssSpellCollateralAction {
     //
     // A table of rates can be found at
-    //    https://ipfs.io/ipfs/QmX2QMoM1SZq2XMoTbMak8pZP86Y2icpgPAKDjQg4r4YHn
+    //    https://ipfs.io/ipfs/QmPgPVrVxDCGyNR5rGp9JC5AUxppLzUAqvncRJDcxQnX1u
```
this revert the correct change (that include the 1bps)

naszam commented on src/Goerli-DssSpell.t.sol:
```diff
@@ -1,6 +1,8 @@
-// SPDX-FileCopyrightText: © 2020 Dai Foundation <www.daifoundation.org>
+// SPDX-FileCopyrightText: © 2021-2022 Dai Foundation <www.daifoundation.org>
```
Diffs: see above

naszam commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -1,6 +1,8 @@
-// SPDX-FileCopyrightText: © 2022 Dai Foundation <www.daifoundation.org>
+// SPDX-FileCopyrightText: © 2021-2022 Dai Foundation <www.daifoundation.org>
```
Diffs: see above

naszam commented on src/test/config.sol:
```diff
@@ -99,9 +99,9 @@ contract Config {
         spellValues = SpellValues({
             deployed_spell:                 address(0),        // populate with deployed spell if deployed
-            deployed_spell_created:         0,                 // use tx=<deploy tx> make deploy-stamp
+            deployed_spell_created:         1657562170,        // use tx=<deploy tx> make deploy-stamp
```
Diffs: this reverts base spell change

naszam commented on src/test/config.sol:
```diff
@@ -99,9 +99,9 @@ contract Config {
+            deployed_spell_created:         1657562170,        // use tx=<deploy tx> make deploy-stamp
             previous_spell:                 address(0),        // supply if there is a need to test prior to its cast() function being called on-chain.
-            office_hours_enabled:           false,             // true if officehours is expected to be enabled in the spell
+            office_hours_enabled:           true,             // true if officehours is expected to be enabled in the spell
```
Diffs: see above

hbarcelos commented:
> Diffs against base branch should be covered now, along with some styling and common patterns related comments.

I think I've addressed all comments so far.
Please let us know if you notice anything else that needs changing.

naszam commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -17,8 +17,66 @@
+import "dss-interfaces/dss/ChainlogAbstract.sol";
+import "dss-interfaces/dss/IlkRegistryAbstract.sol";
+
+interface ERC20Like {
```
Cleanup: redundant, consider using `DSTokenAbstract`

naszam replied:
following below comment, `GemAbstract` could be used instead.

naszam commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -17,8 +17,66 @@
+    function draw(uint256) external;
+}
+
+interface TokenDetailsLike {
```
Consider using erc20 compliant [GemAbstract](https://github.com/makerdao/dss-interfaces/blob/master/src/ERC/GemAbstract.sol)

naszam commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -17,8 +17,66 @@
+}
+
+interface RwaLiquidationLike {
+    function wards(address) external returns (uint256);
```
Cleanup: consider checking and removing unused interfaces

naszam replied:
`wards` is not used in the current spell code, consider removing it and check other unused interfaces, keeping only the ones that are used to reduce code size and ease review.

hbarcelos replied:
Honestly I don't see a point in doing that, because the interfaces for such components are kind of immutable. They shouldn't be a part of the review.

This could be put into `dss-interfaces` like the other `*Abstract` contracts, moved to another file or even moved to the end of the file.

hbarcelos replied:
I created a [PR](https://github.com/makerdao/dss-interfaces/pull/75) to add the MIP21 contracts to `dss-interfaces`.

naszam commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -31,58 +89,277 @@ contract DssSpellCollateralAction {
     //    https://ipfs.io/ipfs/QmTRiQ3GqjCiRhh1ojzKzgScmSsiwQPLyjhgYSxZASQekj
```
Consider updating IPFS hash with latest table of rates

naszam commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -31,58 +89,277 @@ contract DssSpellCollateralAction {
+     *
+     * https://ipfs.io/ipfs/QmdmAUTU3sd9VkdfTZNQM6krc9jsKgF2pz7W1qvvfJo1xk
+     */
+    string constant RWA008_DOC               = "IPFS_HASH"; // Reference to a documents which describe deal (should be uploaded to IPFS)
```
Missing: RWA008 Doc IPFS Hash

hbarcelos replied:
We are still trying to figure out the best way to include the required information on IPFS or similar solution. Since this is a Goerli spell, we believe this document is not too important in order to block a merge. I'll keep you posted.

naszam replied:
Consider using [pinata](https://www.pinata.cloud/) for uploading on IPFS.

hbarcelos replied:
The real issue is that these deals are made of several documents. We are still looking for the authoritative source containing all of them.

I have found a [forum post](https://forum.makerdao.com/t/huntingdon-valley-bank-transaction-documents-on-permaweb/16264?u=miguel_cabeza) that fills that role for HVB and exported its raw content to upload to IPFS.

We are still missing that for SocGen.

naszam commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -31,58 +89,277 @@ contract DssSpellCollateralAction {
+     *
+     * https://ipfs.io/ipfs/QmdmAUTU3sd9VkdfTZNQM6krc9jsKgF2pz7W1qvvfJo1xk
+     */
+    string constant RWA009_DOC               = "IPFS_HASH"; // TODO Reference to a documents which describe deal (should be uploaded to IPFS)
```
Missing: RWA009 Doc IPFS Hash

naszam commented on src/Goerli-DssSpell.t.sol:
```diff
@@ -18,15 +18,82 @@ pragma solidity 0.6.12;
 
-interface DenyProxyLike {
-    function denyProxy(address) external;
+interface ERC20Like {
```
Cleanup: redundant, consider using `GemAbstract` that is imported in base tests

naszam commented on src/Goerli-DssSpell.t.sol:
```diff
@@ -18,15 +18,82 @@ pragma solidity 0.6.12;
-interface StarknetLike {
-    function ceiling() external returns (uint256);
+interface RwaLiquidationLike {
+    function wards(address) external returns (uint256);
```
Cleanup: see above suggestion

naszam commented on src/test/config.sol:
```diff
@@ -1669,5 +1669,65 @@ contract Config {
+            aL_enabled:   false,         // DssAutoLine is enabled?
+            aL_line:      0,             // In whole Dai units
+            aL_gap:       0,             // In whole Dai units
+            aL_ttl:       1 hours,       // In seconds
```
Check previous patterns and adjust parameters accordingly (e.g. `RWA006`)

hbarcelos replied:
Fixed

naszam commented on src/Goerli-DssSpell.t.sol:
```diff
@@ -16,6 +16,7 @@
 
 pragma solidity 0.6.12;
 
+import "dss-exec-lib/DssExecLib.sol";
```
why did you add this? 
`Changelog` is declared in [Goerli-DssSpell.t.base.sol](https://github.com/makerdao/spells-goerli/blob/master/src/Goerli-DssSpell.t.base.sol#L64) and it uses the `chainlog` address specified in `address_goerli.sol`

hbarcelos replied:
I wanted to remove the hard-coded chainlog address from the helper spells.
In the end I removed this import to use the `Addresses` contract instead.

naszam commented on src/Goerli-DssSpell.sol:
```diff
@@ -15,9 +15,9 @@
 
 pragma solidity 0.6.12;
+// Enable ABIEncoderV2 when managing collateral
+pragma experimental ABIEncoderV2;
```
I would keep it commented for now

gbalabasquer commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +77,280 @@ contract DssSpellCollateralAction {
+        // DSS_PAUSE_PROXY permission on URN
+        RwaUrnLike(RWA009_A_URN).hope(address(this));
+
+        // lock RWA009 Token in the URN
+        DSTokenAbstract(RWA009).approve(RWA009_A_URN, 1 * WAD);
+        RwaUrnLike(RWA009_A_URN).lock(1 * WAD);
+
+        // draw DAI to genesis address
+        RwaUrnLike(RWA009_A_URN).draw(RWA009_DRAW_AMOUNT);
```
I'd move this after finishing setting up the RWA to separate what is proper set up (and done just once) than what is minting and will happen more times.

gbalabasquer replied:
Maybe even consider taking it out of the "onboard" function.

hbarcelos replied:
Moved this to the main spell file.

naszam commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +77,280 @@ contract DssSpellCollateralAction {
+        address MCD_SPOT                 = CHANGELOG.getAddress("MCD_SPOT");
+        // --------------------------- RWA Collateral onboarding ---------------------------
+
+        // Onboard SocGen
```
Missing: include poll link

hbarcelos replied:
Done

naszam commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +77,280 @@ contract DssSpellCollateralAction {
+        // Onboard SocGen
+        onboardRwa008(CHANGELOG, REGISTRY, MIP21_LIQUIDATION_ORACLE, MCD_VAT, MCD_JUG, MCD_SPOT);
+
+        // Onboard HvB
```
Missing: include poll link

hbarcelos commented on src/test/addresses_goerli.sol:
```diff
@@ -326,5 +326,21 @@ contract Addresses {
         addr["STARKNET_DAI_BRIDGE"]             = 0xd8beAa22894Cd33F24075459cFba287a10a104E4;
         addr["STARKNET_GOV_RELAY"]              = 0x73c0049Dd6560E644984Fa3Af30A55a02a7D81fB;
         addr["STARKNET_CORE"]                   = 0xde29d060D45901Fb19ED6C6e959EB22d8626708e;
+        addr["RWA_URN_PROXY_ACTIONS"]           = 0x741A0fc23AB643B1963E77B5136554b12a91f170;
```
Done

hbarcelos commented on src/test/config.sol:
```diff
@@ -1669,5 +1669,65 @@ contract Config {
+            aL_enabled:   false,         // DssAutoLine is enabled?
+            aL_line:      0,             // In whole Dai units
+            aL_gap:       0,             // In whole Dai units
+            aL_ttl:       1 hours,       // In seconds
```
Fixed

iamchrissmith commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +56,254 @@ contract DssSpellCollateralAction {
+        /*
+         * init the RwaLiquidationOracle2
+         */
+        // TODO: this should be verified with RWA Team (5 min for testing is good)
```
has this been verified? can the TODO comment be removed?

hbarcelos replied:
Fixed.

gbalabasquer commented on src/Goerli-DssSpell.sol:
```diff
@@ -45,8 +57,26 @@ contract DssSpellAction is DssAction, DssSpellCollateralAction {
+
+    function drawFromRWA009Urn() internal {
+        // DSS_PAUSE_PROXY permission on URN
+        RwaUrnLike(RWA009_A_URN).hope(address(this));
```
I guess this is part of the set up as in RWA008 we set the operator.

gbalabasquer replied:
Yes that is what I mean. In the case of RWA009, the operator is the pause proxy, so you could set it up in the onboarding (as for RWA008) and not in the minting phase.

godsflaw replied:
Looks like you put this in the `onboardRwa008()` section, it should be moved to the `onboardRwa009()` section (tripped me up when I was looking for it).  Lines `165` -> `166`.

gbalabasquer replied:
Yep mentioned it as well in that line

gbalabasquer commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +54,243 @@ contract DssSpellCollateralAction {
+
+    uint256 constant RWA008_A_INITIAL_DC       = 30_000_000 * RAD;
+    uint256 constant RWA008_A_INITIAL_PRICE    = 30_437_069 * WAD;
+    uint48 constant  RWA008_A_TAU              = 0;
```
`tau` is gonna be 0 on mainnet too?

hbarcelos replied:
Yes.

gbalabasquer commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +54,243 @@ contract DssSpellCollateralAction {
+    // MIP21_LIQUIDATION_ORACLE params
+    uint256 constant RWA009_A_INITIAL_DC     = 100_000_000 * RAD;
+    uint256 constant RWA009_A_INITIAL_PRICE  = 100_000_000 * WAD;
+    uint48  constant RWA009_A_TAU            = 0;
```
Same question here.

gbalabasquer commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +54,243 @@ contract DssSpellCollateralAction {
+        VatAbstract(MCD_VAT).file(ilk, "line", RWA009_A_INITIAL_DC);
+        VatAbstract(MCD_VAT).file("Line", VatAbstract(MCD_VAT).Line() + RWA009_A_INITIAL_DC);
+
+        // No dust
+        // VatAbstract(MCD_VAT).file(ilk, "dust", 0)
+
```
```suggestion
```

gbalabasquer commented on src/Goerli-DssSpell.t.sol:
```diff
@@ -167,31 +213,60 @@ contract DssSpellTest is GoerliDssSpellTestBase {
+
+        // RWA TOKEN FAB
+        checkChainlogKey("RWA_TOKEN_FAB");
         // checkChainlogVersion("X.XX.X");
```
Missing this.

hbarcelos replied:
Do you mean the chainlog version test?

gbalabasquer replied:
Right

hbarcelos replied:
Okay, fixing it...

gbalabasquer commented on src/Goerli-DssSpell.t.sol:
```diff
@@ -18,15 +18,61 @@ pragma solidity 0.6.12;
 }
 
 contract DssSpellTest is GoerliDssSpellTestBase {
+    RwaLiquidationLike oracle = RwaLiquidationLike(addr.addr("MIP21_LIQUIDATION_ORACLE"));
+
+    // -- RWA008 --
+    bytes32 constant     ilk_008             = "RWA008-A";
+
+    address              rwaOperator_008     = 0x3F761335890721752476d4F210A7ad9BEf66fb45;
+    DSTokenAbstract      rwagem_008          = DSTokenAbstract(addr.addr("RWA008"));
+    GemJoinAbstract      rwajoin_008         = GemJoinAbstract(addr.addr("MCD_JOIN_RWA008_A"));
+    RwaUrnLike           rwaurn_008          = RwaUrnLike(addr.addr("RWA008_A_URN"));
+    RwaInputConduitLike  rwaconduitin_008    = RwaInputConduitLike(addr.addr("RWA008_A_INPUT_CONDUIT"));
+    RwaOutputConduitLike rwaconduitout_008   = RwaOutputConduitLike(addr.addr("RWA008_A_OUTPUT_CONDUIT"));
+
+    BumpSpellRwa008      bumpSpell_008;
+    TellSpellRwa008      tellSpell_008;
+    CureSpellRwa008      cureSpell_008;
+    CullSpellRwa008      cullSpell_008;
+    EndSpellRwa008       endSpell_008;
+
+    // -- RWA009 --
+    bytes32 constant     ilk_009             = "RWA009-A";
+    DSTokenAbstract      rwagem_009          = DSTokenAbstract(addr.addr("RWA009"));
+    GemJoinAbstract      rwajoin_009         = GemJoinAbstract(addr.addr("MCD_JOIN_RWA009_A"));
+    RwaUrnLike           rwaurn_009          = RwaUrnLike(addr.addr("RWA009_A_URN"));
+    address              RWA009_CES_MULTISIG = addr.addr("RWA009_A_OUTPUT_CONDUIT");
+
+    BumpSpellRwa009      bumpSpell_009;
+    TellSpellRwa009      tellSpell_009;
+    CullSpellRwa009      cullSpell_009;
```
I'd suggest move all this right above the first RWA specific test "testSpellIsCast_RWA00X_INTEGRATION_BUMP".
So for next week we do not forget to remove parts of it.

gbalabasquer replied:
I wouldn't say for this spell but from the next one I think we should have a specific RWA test file.

hbarcelos replied:
Sure.

gbalabasquer commented on src/Goerli-DssSpell.t.sol:
```diff
@@ -458,4 +533,537 @@ contract DssSpellTest is GoerliDssSpellTestBase {
         assertEq(vest.rxd(1), WAD);
     }
+
+    function testSpellIsCast_RWA009_INTEGRATION_BUMP() public {
```
Any reason why not keeping the same 8 then 9 order?

gbalabasquer replied:
I wouldn't say for this spell but from the next one I think we should have a specific RWA test file.

hbarcelos replied:
No particular reason, but I'll reorder them.

godsflaw replied:
resolving...

gbalabasquer commented on src/test/config.sol:
```diff
@@ -98,38 +98,38 @@ contract Config {
         // Values for spell-specific parameters
         //
         spellValues = SpellValues({
-            deployed_spell:                 address(0),        // populate with deployed spell if deployed
```
If you can easily rollback these spacing changes would be appreciated so we can try to reduce to the minimum the code changes.

hbarcelos replied:
I'll do that, however wouldn't it be better if we "fixed" this?

All tools that I know that help automating this (both on Vim or VSCode) will produce the same result by default.

Otherwise we're stuck manually adjusting spaces forever.

gbalabasquer replied:
We use to do PRs for styling fixes, I'd say we discuss what is the proper way through one of them.
It is true that sometimes we also fix styling stuff during new executive PRs, but I'd say that as this is all new process we try to keep restricted to the minimum that needs to change.

gbalabasquer commented on src/test/config.sol:
```diff
@@ -1669,5 +1669,65 @@ contract Config {
+            line:         30_000_000,
+            dust:         0,
+            pct:          5,
+            mat:          10_000,
```
It is confusing putting the "_" separating the three zeros. In this case we are not talking about regular numbers but bps. So it is 100%. I'd just use "10000" like the rest of collaterals or case "100_00" (but we should change everything else which I wouldn't recommend for this PR).

hbarcelos replied:
Fixed.

godsflaw replied:
resolving...

gbalabasquer commented on src/test/config.sol:
```diff
@@ -1669,5 +1669,65 @@ contract Config {
+            line:         100_000_000,
+            dust:         0,
+            pct:          0,
+            mat:          10_000,
```
Same here.

hbarcelos replied:
Fixed.

godsflaw replied:
resolving...

gbalabasquer commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -160,7 +162,9 @@ contract DssSpellCollateralAction {
         GemJoinAbstract(MCD_JOIN_RWA008_A).rely(RWA008_A_URN);
 
-        // set up the urn
+        // DSS_PAUSE_PROXY permission on URN
```
This should go in the RWA009 function.

godsflaw replied:
lol, I just made the same comment.

hbarcelos replied:
Oops, fixed it.

godsflaw commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +56,243 @@ contract DssSpellCollateralAction {
+    address constant RWA009_A_URN            = 0xD2C8588C72026171Ec3a17369ad0f0734E30915d;
+    address constant RWA009_A_JAR            = 0xa484C16D2Ca15706c4B875710d9e80b7F101572B;
+    // Goerli: CES Goerli Multisig / Mainnet: Genesis
+    address constant RWA009_A_OUTPUT_CONDUIT = 0x7a3D23Dc73F7ead55399597aAE6e525b3DF95A88;
```
Out of curiosity, do we have a signed message from the Genesis address in the forums.  I know this one is a CES multisig, but we will want strong confirmation of this address for mainnet.

hbarcelos replied:
Genesis traders don't have access to the signing key to be able to sign arbitrary messages.

We asked them to send DAI to a specific address to prove they own it. I'm looking for the relevant Discord link that document how this happened.

dbkbali replied:
@godsflaw CES completed a test of sending 10DAI to the output conduit account and to demonstrate they had control of this account we asked them to send 10 DAI back . Etherscan transactions here:  https://etherscan.io/tx/0x4fd33ff92d222e6a843e18ce6c2180414dca30dab791f5b00830ee54f069ce81 and https://etherscan.io/tx/0xfecb71d9fa27e4b7b78fe31d0b7778a005ef96412d7d702371d334cab0092f1d  . We were unable to get them to sign a message from that account - because of operational restrictions on private keys which are managed by their custody provider.

gbalabasquer commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +56,243 @@ contract DssSpellCollateralAction {
+        bytes32 ilk = "RWA008-A";
+
+        // Sanity checks
+        require(GemJoinAbstract(MCD_JOIN_RWA008_A).vat() == MCD_VAT, "join-vat-not-match");
```
We could extend the checks for verifying the urn storage values match with the core contracts and the gemJoin itself.

godsflaw replied:
It's funny, we're on the same timeline.  I just thought this 5m ago, then manually verified the addresses anyway.

hbarcelos replied:
It's done!

godsflaw reviewed (commented):
Many of the contracts are deployed with the EOA: [0xb27b6fa77d7fbf3c1bd34b0f7da59b39d3db0f7e](https://goerli.etherscan.io/address/0xb27b6fa77d7fbf3c1bd34b0f7da59b39d3db0f7e)

We try to put our deployer addresses in the `addresses_deployers.sol` to ensure they are `deny()`ed after deployment.  You can see this by looking in the `checkAuth()` function in the test base.  It will double check that anything we tossed into the chainlog has denyed its deployer.

hbarcelos commented:
> Many of the contracts are deployed with the EOA: [0xb27b6fa77d7fbf3c1bd34b0f7da59b39d3db0f7e](https://goerli.etherscan.io/address/0xb27b6fa77d7fbf3c1bd34b0f7da59b39d3db0f7e)
> 
> We try to put our deployer addresses in the `addresses_deployers.sol` to ensure they are `deny()`ed after deployment. You can see this by looking in the `checkAuth()` function in the test base. It will double check that anything we tossed into the chainlog has denyed its deployer.

I add the address there.

gbalabasquer commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +60,252 @@ contract DssSpellCollateralAction {
+        VatAbstract(MCD_VAT).rely(MCD_JOIN_RWA008_A);
+
+        // Allow RwaLiquidationOracle to modify Vat registry
+        VatAbstract(MCD_VAT).rely(MIP21_LIQUIDATION_ORACLE);
```
This permission is already set as is global for all the RWAs. I'd comment it with a message if we want to keep it just for documentation.

hbarcelos replied:
Apparently it's not, at least for Goerli.
If I remove this from both onboarding functions, some tests fail.
I left this in one of them and commented out on the other.

gbalabasquer replied:
Hmm will look into it in more detail. Pretty weird, but I have missed it when we migrated from kovan to goerli.

gbalabasquer commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +60,252 @@ contract DssSpellCollateralAction {
+        RwaLiquidationLike(MIP21_LIQUIDATION_ORACLE).init(ilk, RWA008_A_INITIAL_PRICE, RWA008_DOC, RWA008_A_TAU);
+        (, address pip, , ) = RwaLiquidationLike(MIP21_LIQUIDATION_ORACLE).ilks(ilk);
+
+        // Set price feed for RWA008
```
I've noticed we are not using the `DssExecLib` functions for `file`s, `rely`s.
We have special functions even for debt ceiling, collateralization ratio, fees.
Any reason for that?
RWA001 in Goerli uses all of them.

hbarcelos replied:
Tests are based on [this file](https://github.com/makerdao/MIP21-RWA-Example/blob/master/src/RwaSpell.sol) from MIP21-RWA-Example.
I guess we didn't take the time to update it to use `DssExecLib`.
Will do that now.

gbalabasquer commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +60,252 @@ contract DssSpellCollateralAction {
+            "RWA008-A",
+            MCD_JOIN_RWA008_A,
+            RWA008,
+            GemJoinAbstract(MCD_JOIN_RWA008_A).dec(),
```
```suggestion
            decimals,
```

gbalabasquer commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +60,252 @@ contract DssSpellCollateralAction {
+            "RWA009-A",
+            MCD_JOIN_RWA009_A,
+            RWA009,
+            GemJoinAbstract(MCD_JOIN_RWA009_A).dec(),
```
```suggestion
            decimals,
```

gbalabasquer commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +60,252 @@ contract DssSpellCollateralAction {
+
+        // Add RWA009 to ILK REGISTRY
+        REGISTRY.put(
+            "RWA009-A",
```
```suggestion
            ilk,
```

gbalabasquer commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -28,61 +60,252 @@ contract DssSpellCollateralAction {
+        CHANGELOG.setAddress("RWA008_A_OUTPUT_CONDUIT", RWA008_A_OUTPUT_CONDUIT);
+
+        REGISTRY.put(
+            "RWA008-A",
```
```suggestion
            ilk,
```
